lib/tapioca/dsl/compilers/active_record_associations.rb

```diff
@@ -148,6 +148,20 @@ def gather_constants
 
         private
 
+        ColumnTypeOption = Helpers::ActiveRecordColumnTypeHelper::ColumnTypeOption
+
+        def column_type_option
+          @column_type_option ||= T.let(
+            ColumnTypeOption.from_options(options) do |value, default_column_type_option|
```

Comment:
Yes this would have to be its own option similar to the previous implementation.